### PR TITLE
Fix: upload variables messaging

### PR DESCRIPTION
**Pull request by AbhishekA1509**
# Description
This PR is intended to bring a footer to display error messages in case of validation issues from backend.

Screenshot 2023-09-06 at 7 22 54 PM

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/scopedVariables/SavedVariables.tsx [hunk 1]
```diff
@@ -30,51 +30,64 @@ export default function SavedVariablesView({
...
+
+    const handleSetYAMLView = () => setCurrentView(FileView.YAML)
+
+    const handleSetSavedView = () => setCurrentView(FileView.SAVED)
 
     const onSearch = (query: string) => {
-        const filteredVariables = scopedVariablesData?.spec?.filter((variable) => {
-            return (
+        const filteredVariables = scopedVariablesData?.spec?.filter(
+            (variable) =>
```
Comment (shivani170) [inline src/components/scopedVariables/SavedVariables.tsx:55]: For multiline we can use curly brackets & return

**Review by shivani170 — APPROVED** [review]

File: src/components/scopedVariables/SavedVariables.tsx [hunk 2]
```diff
@@ -30,51 +30,64 @@ export default function SavedVariablesView({
...
 
     useEffect(() => {
         if (status?.status == null) {
-            const variables = scopedVariablesData?.spec?.map((variable) => {
-                return {
-                    name: variable.name,
-                    description: variable.description,
-                }
-            })
+            const variables = scopedVariablesData?.spec?.map((variable) => ({
```
Comment (vivek-devtron) [inline src/components/scopedVariables/SavedVariables.tsx:39]: We should move this check in the if statement and we can omit the next if(variable) check

File: src/components/scopedVariables/ScopedVariablesEditor.tsx [hunk 3]
```diff
@@ -58,10 +56,16 @@ export default function ScopedVariablesEditor({
                 setScopedVariables(null)
                 reloadScopedVariables()
             } else {
-                toast.error(SAVE_ERROR_TOAST_MESSAGE)
+                toast.error(UPLOAD_FAILED_STANDARD_MESSAGE)
             }
         } catch (e) {
-            toast.error(SAVE_ERROR_TOAST_MESSAGE)
+            if (e instanceof ServerErrors && Array.isArray(e.errors) && e?.code === 406) {
+                setFooterError(e?.errors?.[0]?.userMessage || UPLOAD_FAILED_STANDARD_MESSAGE)
```
Comment (vivek-devtron) [inline src/components/scopedVariables/ScopedVariablesEditor.tsx:63]: Please remove unnecessary null checks

File: src/components/scopedVariables/ScopedVariablesEditor.tsx [hunk 4]
```diff
@@ -58,10 +56,16 @@ export default function ScopedVariablesEditor({
...
+                toast.error(UPLOAD_FAILED_STANDARD_MESSAGE)
```
Comment (vivek-devtron) [inline src/components/scopedVariables/ScopedVariablesEditor.tsx:64]: we can remove these 3 lines

File: src/components/scopedVariables/ScopedVariablesEditor.tsx [hunk 5]
```diff
@@ -107,8 +111,20 @@ export default function ScopedVariablesEditor({
         abortRead()
     }
 
+    const renderInfoBarCloseButton = (): JSX.Element => {
+        const handleClearError = () => {
```
Comment (vivek-devtron) [inline src/components/scopedVariables/ScopedVariablesEditor.tsx:115]: We should move this method out

Comment (AbhishekA1509) [reply src/components/scopedVariables/ScopedVariablesEditor.tsx]: Done

File: src/components/scopedVariables/ScopedVariablesEditor.tsx (hunk 4 above)
Comment (AbhishekA1509) [reply src/components/scopedVariables/ScopedVariablesEditor.tsx]: Done

File: src/components/scopedVariables/ScopedVariablesEditor.tsx (hunk 3 above)
Comment (AbhishekA1509) [reply src/components/scopedVariables/ScopedVariablesEditor.tsx]: Done

File: src/components/scopedVariables/SavedVariables.tsx (hunk 2 above)
Comment (AbhishekA1509) [reply src/components/scopedVariables/SavedVariables.tsx]: Done

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**